src/helics/application_api/Federate.hpp

```diff
@@ -73,7 +73,9 @@ class Federate
       asyncCallInfo;  //!< pointer to a class defining the async call information
     std::unique_ptr<FilterFederateManager> fManager;  //!< class for managing filter operations
     std::string name;  //!< the name of the federate
-  public:
+    std::function<void(int, const std::string &, const std::string &)> loggerFunction; //!< Logging function
```

Comment:
This should probably use the setLoggingCallback method on the core instead of storing it separately.  That way any log messages from the internal federate would get logged as well and make sure things are synchronized properly.

Reply:
This was also my first inclination, however doing so resulted in a series of behavior problems that I was ultimately unable to resolve without changing components in the Core and CommonCore which I was not familiar enough with to change (possibly big breaking components). 
The trace using the setLoggingCallback and passing messages through the core ultimate reaches a point of failure at CommonCore.cpp:1788 when that path of logic fails to identify as a federate, gets a nullptr, and returns false, never logging the message. 
This may itself be a bug, rather than a configuration issue, however I was unable to resolve it in a way that I found satisfactory.

Reply:
I think we should revisit that and if there is some bugs we should fix them.  The way it is setup in this PR is that only the messages that get sent from the new federate API calls will end up in that log, and I don't think that is really what we intend, we really want an ability to merge the core generated message with user generate ones, to get a good sequence.